src/cls/ZPM/Analytics/Event.cls

```diff
@@ -0,0 +1,112 @@
+Class ZPM.Analytics.Event Extends (%Persistent, %JSON.Adaptor)
+{
+
+/// Server date and time, saving the data 
+Property TS As %PosixTime(%JSONINCLUDE = "none") [ SqlComputeCode = {set {*}=##class(%Library.PosixTime).CurrentTimeStamp()}, SqlComputed, SqlComputeOnChange = %%INSERT ];
+
+/// Event type: download, install, uninstall
+Property Type As %String(%JSONINCLUDE = "none", VALUELIST = ",download,install,uninstall");
+
+/// ModuleName
+Property Name As %String(%JSONFIELDNAME = "name", MAXLEN = 50) [ Required ];
+
+Property Version As %String(%JSONFIELDNAME = "version", MAXLEN = 50) [ Required ];
+
+Property GUID As %String(%JSONFIELDNAME = "guid");
+
+/// Iris Version ($zv)
+Property SystemVersion As %String(%JSONFIELDNAME = "system_version", MAXLEN = 1000);
+
+Property SystemVersionOS As %String(%JSONINCLUDE = "none") [ SqlComputeCode = {set {*}=##class(%SYSTEM.Version).GetOS({SystemVersion})}, SqlComputed, SqlComputeOnChange = %%INSERT ];
+
+Property SystemVersionNumber As %String(%JSONINCLUDE = "none") [ SqlComputeCode = {set {*}=##class(%SYSTEM.Version).GetNumber({SystemVersion})}, SqlComputed, SqlComputeOnChange = %%INSERT ];
+
+Property ZPMClientVersion As %String(%JSONFIELDNAME = "zpmclient_version", MAXLEN = 50);
+
+Property IP As %String(%JSONINCLUDE = "none", MAXLEN = 50);
```

Comment:
I would suggest collecting IP address only in case if it works in a [private network](https://en.wikipedia.org/wiki/Private_network). So, collect it only for a few subnets.

The global version does need it, we may use GeoIP, to collect territories.